### PR TITLE
[feat] style settings page

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Add styling to `/settings` page.

Add component `SettingsSection` to handle each card section independently.

Add `BackLink` component (a part of `/styles/text.tsx`)

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

settings_page_styled

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

#### Code Review

Recommended order:
- `src/app/settings`
    - `/index.tsx`
    - `/styles.ts`
- `src/app/components/SettingsSection`
    - `/index.tsx`
    - `/styles.ts`

#### Testing

Standard procedure:
```
git fetch origin jinkang/ijp-81-style-settings-page
git checkout jinkang/ijp-81-style-settings-page
npm run dev
```

Navigate to `/settings` to see

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Add functionality in a later sprint.

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar, @davidqing6432

### CODE REVIEW

File: src/app/settings/page.tsx [hunk 1]
```diff
@@ -1,29 +1,110 @@
...
-      <button type="button" onClick={resetPassword}>
-        Reset Password
-      </button>
-    </>
+    <PageContainer>
+      <BackLink href="/cases">Back</BackLink>
+      <ContentContainer>
+        <H1>Your Profile</H1>
+
+        <SettingsSection
```
Comment (pragyakallanagoudar) [inline src/app/settings/page.tsx:33]: let's remove the edit icon from here as per the figma -- reset email/password from settings is out of scope / potential stretch for rn

File: src/app/settings/page.tsx [hunk 2]
```diff
@@ -1,29 +1,110 @@
...
-      <button type="button" onClick={handleSignOut}>
-        Sign out
-      </button>
-      <button type="button" onClick={resetPassword}>
-        Reset Password
-      </button>
-    </>
+    <PageContainer>
+      <BackLink href="/cases">Back</BackLink>
+      <ContentContainer>
```
Comment (pragyakallanagoudar) [inline src/app/settings/page.tsx:30]: @davidqing6432 curious if we can move the entire settings column up so it's closer to the top of the page? i.e. at the level of the back button maybe

they feel pretty far away from each other rn, which is chill but my only concern is that you can scroll a tiny bit and the screen still looks normal but the back button is gone

Screenshot 2023-11-23 at 8 13 13 PM

File: src/styles/text.tsx [hunk 3]
```diff
@@ -52,3 +52,45 @@ export const LinkColored = styled(Link)<{ $color?: string }>`
   ${openSans.style}
   color: ${props => props.$color || COLORS.greyDarker};
 `;
+
+export const BackLink = styled(Link)`
```
Comment (pragyakallanagoudar) [inline src/styles/text.tsx:56]: this is SO CLEANNNNN

File: src/components/SettingsSection/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+// helpers
+const chooseFormatter = (data: SectionData): string => {
+  if (data.format) return data.format(data.value);
+  return data.value instanceof Array
+    ? data.value.join(', ')
+    : data.value.toString();
+};
+
+const dataMapper = (data: SectionData | SectionData[] | string) => {
```
Comment (pragyakallanagoudar) [inline src/components/SettingsSection/index.tsx:32]: this is INCREDIBLEEE such good intuition here!!

**Review by pragyakallanagoudar — COMMENTED** [review]
ur crazy for this....it looks FANTASTICCCC

have some nits and design questions— some are on here, and some are on figma (sorry if that's confusing, i can also just add the figma ones on here too if you'd prefer). but overall AMAZING WORK

File: src/app/settings/page.tsx (hunk 2 above)
Comment (david-qing) [reply src/app/settings/page.tsx:30]: yeah we can move it up!!

**Review by pragyakallanagoudar — APPROVED** [review]
LOOKING GOODD!! the header spacing is perfect now, think this is ready to merge in

**Pull request merged**